app/src/parser.c

```diff
@@ -297,6 +297,48 @@ parser_error_t parser_printArgumentPublicKeys(const parser_context_t *argumentCt
     return PARSER_OK;
 }
 
+parser_error_t parser_printArgumentOptionalPublicKeys(const parser_context_t *argumentCtx, uint8_t argumentIndex,
+                                              char *outVal, uint16_t outValLen,
+                                              uint8_t pageIdx, uint8_t *pageCount) {
+    MEMZERO(outVal, outValLen);
+
+    parsed_json_t parsedJson = {false};
+    CHECK_PARSER_ERR(json_parse(&parsedJson, (char *) argumentCtx->buffer, argumentCtx->bufferLen));
+
+    // Estimate number of pages
+    uint16_t internalTokenElementIdx;
+    CHECK_PARSER_ERR(json_matchOptionalArray(&parsedJson, 0, &internalTokenElementIdx));
+    if (internalTokenElementIdx == JSON_MATCH_VALUE_IDX_NONE) {
+        if (outValLen < 5) {
+            return  PARSER_UNEXPECTED_BUFFER_END;
+        }
+        *pageCount = 1;
+        strncpy_s(outVal, "None", 5);
+    }
+    else {
+        uint16_t arrayTokenCount;
+        CHECK_PARSER_ERR(array_get_element_count(&parsedJson, internalTokenElementIdx, &arrayTokenCount));
+        if (arrayTokenCount > 64) {
```

Comment:
:thinking: do we have some intuition for this check? It seems a bit arbitrary

Reply:
Hmm. This is copy-paste from non-optional function. As far as I see, the constant 64 is really quite random there too and as of now it limits indirectly the number of Public Keys in Create Account transaction. We plan to limit this to 3 in the future. I do not want to mess with Create Account transaction now.
For now, I just introduced a new define which puts these two related constants together and I added a short comment on what the test does. This particular test will probably change very soon (when I will be confident about the state of testing of Create Account transaction).